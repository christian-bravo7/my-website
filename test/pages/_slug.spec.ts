import Slug from '@/pages/_slug.vue';
import PortalVue from 'portal-vue';
import VueMeta from 'vue-meta';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(PortalVue);
localVue.use(VueMeta, { keyName: 'head' });

describe('single post page', () => {
  it('should render page correctly', () => {
    const wrapper = mount(Slug, {
      localVue,
      stubs: {
        NuxtContent: {
          template: '<div></div>',
        },
      },
      data: () => ({
        articles: {
          slug: 'hello-world',
          title: 'Hello world',
          description: 'Welcome to my first post of the many that I hope to share us ❤️',
          minutes: 2,
          image: 'https://images.pexels.com/photos/2646531/pexels-photo-2646531.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        },
      }),
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display meta-tag title', () => {
    const wrapper = mount(Slug, {
      localVue,
      stubs: {
        NuxtContent: {
          template: '<div></div>',
        },
      },
      data: () => ({
        articles: {
          slug: 'hello-world',
          title: 'Hello world',
          description: 'Welcome to my first post of the many that I hope to share us ❤️',
          minutes: 2,
          image: 'https://images.pexels.com/photos/2646531/pexels-photo-2646531.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        },
      }),
    });

    const title = wrapper.vm.$meta().refresh().metaInfo.title;

    expect(title).toBe('Hello world');
  });
});
