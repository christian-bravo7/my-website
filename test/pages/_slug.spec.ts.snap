// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single post page should render page correctly 1`] = `
<section>
  <div class="v-portal" style="display: none;"></div>
  <div>
    <div class="container mx-auto py-20 px-4">
      <div document="[object Object]" class="blog-content"></div>
    </div>
  </div>
</section>
`;
